KEP-4008: CRDValidationRacheting - Beta Docs (#45264)
kubernetes/website: content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md

```
### Validation ratcheting

> {{< feature-state state="beta" for_k8s_version="v1.30" >}}

Reviewer: ```suggestion
`{{< feature-state feature_gate_name="CRDValidationRatcheting" >}}`
```

(Less feature state to manage)

Reviewer 2: Yes, but without the backticks

Reviewer: heh, suggestion updated

> You need to enable the `CRDValidationRatcheting`
> [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) to
> use this behavior, which then applies to all CustomResourceDefinitions in your
> cluster. Starting in 1.30, `CRDValidationRatcheting` is enabled by default.

Reviewer 2: ```suggestion
If you are using a version of Kubernetes older than v1.30, you need to explicitly
enable the `CRDValidationRatcheting`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) to
use this behavior, which then applies to all CustomResourceDefinitions in your
cluster.
```

```
### Validation ratcheting

{{< feature-state feature_gate_name="CRDValidationRatcheting" >}}

If you are using a version of Kubernetes older than v1.30, you need to explicitly
enable the `CRDValidationRatcheting`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) to
use this behavior, which then applies to all CustomResourceDefinitions in your

> cluster. f you are using a version of Kubernetes older than v1.30, 

Reviewer 3: Seems like this line has been repeated.
```suggestion
cluster.
```

> you need to explicitly enable the `CRDValidationRatcheting`

Reviewer 3: Repeated content.
```suggestion
```

> [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) to

Reviewer 3: Repeated line.
```suggestion
```

> use this behavior, which then applies to all CustomResourceDefinitions in your

Reviewer 3: ```suggestion
```

> cluster.

Reviewer 3: ```suggestion
```

  To remove a previously specified `additionalProperties` validation will not be
  ratcheted.
- `metadata`
  Errors arising Kubernetes' built-in validation of an object's `metadata` are not

>   ratcheted (such as name, label format, etc.), but user=sepcified rules on metadata 

Reviewer 2: We [avoid Latin phrases](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-latin-phrases) and their abbreviations.
```suggestion
  ratcheted (such as object name, or characters in a label value). If you specify your own additional rules
  for the metadata of a custom resource, that additional validation
```

Author: The linked doc mentions an exception for etc., is it okay to use that here?

> Exception: Use "etc." for et cetera

Reviewer 2: D'oh. That one is OK. I'd still fix the typo (“sepcified”) and reword along the lines of my suggestion, though.

#### The `optionalOldSelf` field {#field-optional-oldself}

> {{< feature-state state="beta" for_k8s_version="v1.30" >}}

Reviewer 2: ```suggestion
{{< feature-state feature_gate_name="CRDValidationRatcheting" >}}
```

> The feature [CRDValidationRatcheting](#validation-ratcheting) must be enabled in order to 
> make use of this field. If the feature is not enabled, this field is ignored.

Reviewer 2: This is what I expected to see:
```suggestion
If your cluster does not have [CRD validation ratcheting](#validation-ratcheting) enabled, 
the CustomResourceDefinition API doesn't include this field, and trying to set it may result
in an error.
```
